package.json

```diff
@@ -205,6 +205,7 @@
     "node-forge": "^1.3.1",
     "omelette": "^0.4.17",
     "open": "^8.4.0",
+    "rss-parser": "^3.12.0",
```

Comment:
@Jwaegebaert we rechecked this package internally with @waldekmastykarz and @martinlingstuyl and we rather don't like the fact that this npm package (and it's dependencies) is inactive for a year time or sometimes longer.
Also since rss is a valid xml we may use the `@xmldom/xmldom` to parse it which is a dependency we already use.
I checked it quickly with a small prototype and seems we are able to parse it as an object
![image](https://user-images.githubusercontent.com/58668583/180667887-c336350e-e383-4568-bef8-23713695a0eb.png)
Could we recheck if this package is actually needed? Do you think it will be hard to 'live without it' in this case, so we may avoid maintaining additional dependency 😉. Let me know if you want do check this case further 🙂

Reply:
BTW @pnp/cli-for-microsoft-365-maintainers open for any additional comments on this one 👍

Reply:
Do we possibly have some logic that is able to convert XML to a json because that is the main reason I'm making use of this package. Don't really know what the best practice is for this scenario.

Reply:
Hi @Jwaegebaert, not as far as I know, but you could check what functionality the xmldom package has to offer. If nothing is available, there's no other route but iterate through the dom-nodes and build the json yourself.

Reply:
Can't seem to get the DOMParser to work for me.

``` typescript
const requestOptions: any = {
  url: `https://developer.microsoft.com/en-us/graph/changelog/rss${searchParam}`,
  headers: {
    'accept': 'text/xml',
    'x-anonymous': 'true'
  }
};

request
  .get(requestOptions)
  .then((output: any) => {
    logger.log(output.toString());
    const parser = new DOMParser();
    const xmlDoc = parser.parseFromString(output.toString(), "text/xml");
    //const changelog: Changelog = this.filterThroughOptions(args.options, this.mapChangelog(output));
    
    logger.log(xmlDoc);
    cb();
  });
```

When I run the following, I keep stumbling upon this error:

![afbeelding](https://user-images.githubusercontent.com/38426621/183676052-baa1e71b-5b1a-46cf-84cf-78049294f06b.png)

Reply:
@Jwaegebaert thanks for reaching out. I will try to have a look at this in the coming days 👍

Reply:
@Jwaegebaert I think the error you posted is actually coming from `logger.log()` rather then from the `parseFromString` (this seems to be working fine) 🤔.
I think the problem is we should adapt (map) the xmldom to a json object. For example to the `Changelog` type you added. It's been some time since I last parsed xml nodes so maybe there are some better (more fancy 😉) ways to go over xml nodes but the general idea is that after we parse the string to the xml we then need to map it to a JSON kind of like object and then proceed working with it. 
```javascript
request
      .get(requestOptions)
      .then((output: any) => {

        const xmlDoc = parser.parseFromString(output.toString(), "text/xml");

        // ------ 👇 new code 👇 ------
        const channel = xmlDoc.getElementsByTagName('channel').item(0);
        const changelog: Changelog = {
          title: channel?.getElementsByTagName('title').item(0)?.textContent ?? '',
          description: channel?.getElementsByTagName('description').item(0)?.textContent ?? '',
          url: channel?.getElementsByTagName('link').item(0)?.textContent ?? '',
          items: []
        } as Changelog;

        Array.from(xmlDoc.getElementsByTagName('item')).forEach((item: any) => {
          changelog.items.push({
            guid: item.getElementsByTagName('guid').item(0).textContent,
            title: item.getElementsByTagName('title').item(0).textContent,
            description: item.getElementsByTagName('description').item(0).textContent,
            pubDate: new Date(item.getElementsByTagName('pubDate').item(0).textContent)
          } as ChangelogItem);
        });

        // ------ 👆 new code 👆 ------
        
        logger.log(changelog);
        cb();
      });
```

the above gives me a huge result which is hard to read but seems to have what was intended.
![image](https://user-images.githubusercontent.com/58668583/184713869-c942bd31-f3d5-4255-94d0-37d43b82b541.png)


let me know if this was of any help 🤔?

Reply:
Thanks @Adam-it! I will give it a shot later today.

Reply:
Thanks again @Adam-it, works like a charm 😄